bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -370,6 +372,27 @@ protected Point computeSize (int part, int state, GC gc, int wHint, int hHint) {
 		return new Point(width, height);
 	}
 
+	/**
+	 * Returns padding for the text of a tab when image is not available or is hidden.
+	 *
+	 * @param item CTabItem
+	 * @param state current state
+	 *
+	 */
+	private int getTextPadding(CTabItem item, int state) {
+		CTabFolder parent = item.getParent();
+		Image image = item.getImage();
+		String text = item.getText();
+
+		if (text != null && parent.getMinimumCharacters() != 0) {
+			if (image == null || image.isDisposed() || ((state & SWT.SELECTED) != 0 && !parent.showSelectedImage)
+					|| ((state & SWT.SELECTED) == 0 && !parent.showUnselectedImage))
+				return INTERNAL_SPACING;
```

Comment:
See general comment: I propose to define the padding when not showing icons separately.
```suggestion
				return TABS_WITHOUT_ICONS_PADDING;
```